pkg/metrics/collector/sockstat.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2018 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type sockStatCollector struct {
+	socketInUseDesc    *prometheus.Desc
```

Comment:
same as above

Reply:
Accidentally hit resolve for the comment above, so will reply here. Understand your points in general. I would think one advantage to make them local is that when they are disabled, no global object would be created. If you still want to make them global, let me know, I can update it.

Reply:
global object should not be a problem...

Reply:
OK, updated to using global vars.